[MENFORCER-411] DependencyConvergence takes include/exclude parameters to filter errors

fdfea opened the pull request:
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

slawekjaranowski commented on enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/utils/DependencyVersionMap.java:
```diff
@@ -120,16 +122,89 @@ private boolean containsConflicts( List<DependencyNode> nodes )
         List<List<DependencyNode>> output = new ArrayList<>();
         for ( List<DependencyNode> nodes : idsToNode.values() )
         {
+            if ( includes != null && !includes.isEmpty() )
```
Why not check `formattedIncludes` - `includes` is not used in this code block

slawekjaranowski commented on maven-enforcer-plugin/src/it/projects/dependency-convergence_exclude_failure/invoker.properties:
```diff
@@ -0,0 +1,18 @@
+# specific language governing permissions and limitations
+# under the License.
+
+invoker.buildResult=failure
```
missing new line 😄

slawekjaranowski replied:
missing new line 😄

slawekjaranowski reviewed (commented):
@fdfea Thanks for your PR

 - simple comments to code
 - I will happy to see options with only `excludes` without declaring empty `includes` 
  - please rebase code with the latest master 
   
Please select one of checkbox about licence to acknowledge your contribution under AFS licence

chicobento commented:
Hi, I see that this hasnt been changed in a long time. Are there any plans on having this brought up again and possibly merged ? Im specially looking into ways of excudling tests deps from the dep convergence analysis and I hope that this change would be of great help.

fdfea commented:
Hi @slawekjaranowski , I forgot about this PR. Thanks for taking a look at it. I've addressed your comments and rebased with master. Please take another look when you get a chance. 

Do I need to sign a contributor agreement for this or will checking the "I hereby declare..." checkbox suffice?

slawekjaranowski commented:
Hi @fdfea please rebase and resolve conflicts.

fdfea commented:
@slawekjaranowski Updated

jira-importer commented:
Resolve #715

jira-importer commented:
Resolve #715
